Update key-metrics.mdx to add runtime metrics (#29434)
hashicorp/vault: website/content/docs/internals/telemetry/key-metrics.mdx

## Resource metrics

### Runtime metrics

#### Metrics:

`vault.runtime.num_goroutines`

`vault.runtime.heap_objects`

#### Background:

> A fundamental part of Vault's runtime are [goroutines](https://go.dev/tour/concurrency/1), which are lightweight threads managed by the Go runtime. Many functions within Vault create goroutines as part of the runtime. We recommend monitoring key metrics of the runtime to establish an accurate baseline and thresholds for alerting on the health of your Vault installation. 

Reviewer: ```suggestion
A fundamental part of the Vault runtime are
[`goroutines`](https://go.dev/tour/concurrency/1). `goroutines` are lightweight
threads managed by the Go runtime that many functions within Vault create as
part of the runtime.

We recommend monitoring key metrics of the runtime to establish an accurate
baseline and alerting thresholds for the health of your Vault installation. 
```

Style correction: avoid possessives

#### Alerting:

> Two key metrics to track are `vault.runtime.num_goroutines` and `vault.runtime.heap_objects`. If you notice a sudden increase in the `vault.runtime.num_goroutines` metric this could affect system load and warrant investigation. A good indicator of memory pressure is `vault.runtime.heap_objects`. Having an accurate baseline and alert threshold for this metric along with monitoring and alerting on memory usage on Vault nodes will help identify potential performance issues.

Reviewer: ```suggestion
Two key alerting metrics to track are `vault.runtime.num_goroutines` and
`vault.runtime.heap_objects`:

- Sudden increases to `vault.runtime.num_goroutines` can indicate there is
  something affecting system load warrant investigation.

- Changes to `vault.runtime.heap_objects` can indicate memory pressure.

Having accurate baselines and alert thresholds for `vault.runtime.heap_objects`
and  `vault.runtime.heap_objects` can help you identify potential performance
issues before they become a problem.
```

Style correction: avoid "this" as a pronoun, make key information easy to see at a glance
